src/testoptix/stringtable.h

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright (c) 2009-2018 Sony Pictures Imageworks Inc., et al.
+All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+* Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+* Neither the name of Sony Pictures Imageworks nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOTSS
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#pragma once
+
+#include <map>
+#include <utility>
+
+#include <OpenImageIO/ustring.h>
+#include <OSL/oslexec.h>
+
+#include <cuda_runtime_api.h>
+#include <optix_world.h>
+
+
```

Comment:
Should this class be in namespace OSL? (Less important if it's just for testoptix, more important if it's going to be "exported" later to be a utility useful for any renderer using OptiX+OSL.)

Reply:
I think StringTable should go into OSL namespace if and when it goes into OSL proper (vs textoptix).  Which, as you say, we can do later.  For now it's in testoptix which is presumably "copy and modify" code for anyone using OptiX+OSL.

Reply:
StringTable is kind of a minimal working example. It needs a bit of work to reach library-grade status.

Reply:
Gotcha. Fine, then. I was thinking that if it's substantially what any OptiX renderer would need, we would want them to use it, not duplicate it. But if it's a minimal example, that's fine. What else would a real renderer want from this class?